### DISCUSSION TITLE
DistributedBoxCollection node based cell simulation domain error

### DISCUSSION THREAD

**Discussion by Elphikaus** [Chaste/Chaste#183] [Q&A] [answered]
I am  running a cell simulation with a node-based off-vertex model.

I set some nodes and run my simulation for 10 hours with a cell cycle uniform 1 hour. 

After some time I get this failed test error: The point provided is outside all of the boxes

From this block of code:
```
template<unsigned DIM>
unsigned DistributedBoxCollection<DIM>::CalculateContainingBox(c_vector<double, DIM>& rLocation)
{
    // The node must lie inside the boundary of the box collection
    for (unsigned i=0; i<DIM; i++)
    {
        if ((rLocation[i] < mDomainSize(2*i)) || !(rLocation[i] < mDomainSize(2*i+1)))
        {
            EXCEPTION("The point provided is outside all of the boxes");
        }
    }
```

When I run the same test for shorter periods of time, the error goes away.

I think this means that the box does not increase as the population increases.

Is this the case?

**Comment by jmpf** [comment, marked as answer]
Apologies -- I don't recall seeing this discussion or issue before.

If this is related to running a node-based simulation for a longer period of time then, yes, the cell population will eventually expand to more than the box size.  While there are ways of reallocating the box-collection the simplest thing to do is to estimate and allocate sufficient space before the simulation.

Just in case it's related I want to point out that there was a bug in the DistributedBoxCollection which I fixed last year.  See #95 for detail -- but it was essentially a bad floating-point fudge which caused a similar exception _before_ the simulation started.
